lib/kubeclient/http_proxy_wrapper.rb

```diff
@@ -0,0 +1,49 @@
+module Kubeclient
+  module Common
+    # Wrapper around Net:HTTP to support k8s services
+    class HttpProxyWrapper
```

Comment:
Can you please specify what is the use case for post to those proxied services?
If both get and post are relevant, then some tests seem missing.